sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -123,7 +123,11 @@ object JavaTypeInference {
         val beanInfo = Introspector.getBeanInfo(typeToken.getRawType)
         val properties = beanInfo.getPropertyDescriptors.filterNot(_.getName == "class")
```

Comment:
I am sorry for the long comment that maybe a bit messed around. So, there are two code paths

- `Encoders.bean(...)`: non-empty required OK & getter/setter required

- `JavaTypeInference.inferDataType`: empty one OK & getter only OK

Could I please ask which case you mean just to double check?